src/Support/Metrics/Trend.php

```diff
@@ -0,0 +1,215 @@
+<?php
+
+namespace FluxErp\Support\Metrics;
+
+use Carbon\CarbonImmutable;
+use Carbon\CarbonInterface;
+use Carbon\CarbonPeriod;
+use FluxErp\Enums\TimeFrameEnum;
+use FluxErp\Support\Metrics\Results\Result;
+use Illuminate\Support\Str;
+use InvalidArgumentException;
+
+class Trend extends Metric
+{
+    protected string $unit;
+
+    public function __call($method, $parameters)
```

Comment:
please refactor this to not use the __call() method.
My suggestion would be:
- add methods for every aggregate (min, max, sum, etc.)
- each of those has 2 parameters, string $column, and optional parameter string $unit
- if no $unit is null, use $this->getRange() to try to get the unit. (this can be done in setType() method to reduce duplicate code)